src/system.rs

```diff
@@ -220,10 +248,27 @@ impl PendingTypes {
                 TypeDefinition::Named(named_type_def) => {
                     match isl_imported_type_name {
                         None => {
+                            // check if the type definitions that are not yet resolved actually exists within the schema
+                            // we can use the isl_type_names to make sure if they exists, otherwise return error.
+                            if named_type_def.is_deferred_type_def()
+                                && !isl_type_names
+                                    .contains(named_type_def.name().as_ref().unwrap().as_str())
+                            {
+                                return unresolvable_schema_error(
+                                    "Unable to load schema due to unresolvable type",
+                                );
+                            }
                             // imports all types into imported_type_ids_by_name section of type_store
                             type_store.add_isl_imported_type(None, named_type_def);
                         }
                         Some(import_type_name) => {
+                            // if there are any deferred type definitions for this import
+                            // then return an error
+                            if named_type_def.is_deferred_type_def() {
+                                return unresolvable_schema_error(
+                                    "Type definition being imported has dependency to a type reference which is not being imported from import schema",
```

Comment:
I don't understand what this message means. Can you explain to me how would an `ion-schema-rust` user cause this error to occur?

Reply:
I meant say that if ta type definition that is being imported has a type reference that is not imported then that should return an error. 
Example: 
```rust
             (
                "sample.isl",
                r#"
                    schema_header::{
                      imports: [ { id: "sample_builtin_nullable_types.isl", type: my_text } ],
                    }
                    
                    type::{
                      name: my_type,
                      type: my_text
                    }
                    
                    schema_footer::{
                    }
                "#,
            ),
            (
                "sample_builtin_nullable_types.isl",
                r#"
                    schema_header::{
                      imports: [],
                    }
                    
                    type::{
                      name: my_text,
                      one_of: [
                        my_string, // <--- this type reference was not improted in sample.isl
                        symbol,
                      ],
                    }
                    
                    type::{
                        name: my_string,
                        type: string,
                    }
                    
                    schema_footer::{
                    }
                "#,
            ),
 ```

Reply:
I will add type definition name for the type import and transitive type def in this error message.

Reply:
Ah, but this example you provided should be valid. The `sample.isl` schema should not have to import `my_string` because `my_string` is abstracted behind `my_text`. If ISL required you to import even types that you were using indirectly, it would quickly become unmaintainable to logically organize types across multiple files.